Lib/unittest/mock.py

```diff
@@ -2482,6 +2485,52 @@ def __set__(self, obj, val):
         self(val)
 
 
+class WaitableMock(MagicMock):
+    """
+    A mock that can be used to wait until it was called.
+
+    `event_class` - Class to be used to create event object.
+    Defaults to Threading.Event and can take values like multiprocessing.Event.
+    """
+
+    def __init__(self, *args, event_class=threading.Event, **kwargs):
+        _safe_super(WaitableMock, self).__init__(*args, **kwargs)
+        self._event_class = event_class
+        self._event = event_class()
+        self._expected_calls = defaultdict(event_class)
+
+    def _mock_call(self, *args, **kwargs):
+        ret_value = _safe_super(WaitableMock, self)._mock_call(*args, **kwargs)
+
+        for call in self._mock_mock_calls:
+            event = self._expected_calls[call.args]
+            event.set()
+
+        self._event.set()
+
+        return ret_value
+
+    def wait_until_called(self, timeout=1.0):
```

Comment:
Please remove default timeout. There is no such "good default" timeout.

Reply:
I don't know. 1.0 seconds seems like a good timeout to me.

Reply:
Agreed. It also prolongs the test suite as more tests are added. @mariocj89  would be creating `EventMock` superseding the PR. Would be happy to know his approach on asserting events and timeouts.

Reply:
IMHO it's a bad idea to hardcode a default timeout. We cannot make assumption on the hardware performance. Please have a look at https://bugs.python.org/issue36369 : Python is slow on a Raspberry Pi Zero W. Does it surprise anyone? Why would Python expect that it's always run on fast x86-64?

I suggest to either block by default (make the timeout optional) or to require a timeout value.

In the worst case, please make at least the default configurable.

I spent like 5 years to fix the Python test suite because too many tests used hardcoded timeouts which fit well for a fast desktop computer, but not for our slowest buildbot workers. Get a test failure only because the timeout is too short is annoying.